### DISCUSSION TITLE
News for PowMR hybrid 10.2Kw inverter

### DISCUSSION THREAD

**Discussion by YAYASIMPSON** [softwarecrash/Solar2MQTT-ESP8266#83] [General]
hello everyone, this post is to announce that Solar2mqtt works wonderfully with the PowMr 10.2Kw hybrid inverter and Jeedom (a French home automation system) thank's for this github

**Comment by softwarecrash** [comment]
Thank you for the Report, it will be added

**Comment by YAYASIMPSON** [comment]
is it possible tonhave full of data because it's okay but a good part of the possible data for the powmr is missing, you can see here. thank's
[POW-HVM5_1_22_translate RS232(2023-09-14 05_44_28).docx](https://github.com/softwarecrash/Solar2MQTT/files/14972165/POW-HVM5_1_22_translate.RS232.2023-09-14.05_44_28.docx)

**Reply by softwarecrash** [reply]
This is modbus rtu communication and Not supported by solar2mqtt at the moment

**Comment by AlexK98** [comment]
@YAYASIMPSON can you share somewhere your experience making solar2mqtt work with PowMr 10.2. My 10.2 (version 79.06, with new 'sci-fi' display) is not recognized at all.

**Reply by AlexK98** [reply]
I made it.
The issue was in wrong RX/TX connection wires order - switched wires and it works now.
Using 2.0.0-Pre firmware.
![2024-12-07 19 14 24](https://github.com/user-attachments/assets/3118944c-3c5d-4fc8-8abf-25fc6ff829ef)

But, in MQTT Exprorer, The inverter is reported as "VMII-NXPW5KW", but mine is 10.2kWt model.
![2024-12-07 19 15 25](https://github.com/user-attachments/assets/c8ff7016-9046-4b9a-bb36-cad282e3be0c)

**Reply by softwarecrash** [reply]
> But, in MQTT Exprorer, The inverter is reported as "VMII-NXPW5KW", but mine is 10.2kWt model. ![2024-12-07 19 15 25](https://private-user-images.githubusercontent.com/20613948/393536607-c8ff7016-9046-4b9a-bb36-cad282e3be0c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1OTIyNDUsIm5iZiI6MTczMzU5MTk0NSwicGF0aCI6Ii8yMDYxMzk0OC8zOTM1MzY2MDctYzhmZjcwMTYtOTA0Ni00YjlhLWJiMzYtY2FkMjgyZTNiZTBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA3VDE3MTkwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkZGVlZDU1Nzk0MGY5Nzk4NTI1Y2NlZjQ4ZDM0NTUwNWEzNWRlMjhhNWRiYjFhOWJmZTg2YWRmYTJkNzMwYjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JJYhF4FmVJiRak9g4WVVrvX94rnx3iWyycmrigRsAcI)

that is was the device gives out

**Reply by AlexK98** [reply]
> that is was the device gives out

@softwarecrash :) you did a good work!

btw, in LiveData one of the params is missing a name:
![2024-12-07 19 38 06](https://github.com/user-attachments/assets/a482a11f-c9a8-4024-96e6-d9fb31cd4438)

**Reply by softwarecrash** [reply]
> > that is was the device gives out
> 
> @softwarecrash :) you did a good work!
> 
> btw, in LiveData one of the params is missing a name: ![2024-12-07 19 38 06](https://private-user-images.githubusercontent.com/20613948/393537869-a482a11f-c9a8-4024-96e6-d9fb31cd4438.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1OTM2NjMsIm5iZiI6MTczMzU5MzM2MywicGF0aCI6Ii8yMDYxMzk0OC8zOTM1Mzc4NjktYTQ4MmExMWYtYzlhOC00MDI0LTk2ZTYtZDlmYjMxY2Q0NDM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA3VDE3NDI0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVhMDdiNDdkNDVjNjJjMjQ4MDA2YWRjYTc0YTQxYjkxYTAwNDk4NGQ3MDkyMjgwY2QwMWIwY2U1N2Q2YzY4ODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ziiyGybRhDpOhMMuI9XN-ZP5XNW_ZnG5xt8km3qAHs8)

you are in the pre relase variants, that have some errors,
can you copy the raw data

**Reply by AlexK98** [reply]
> you are in the pre relase variants, that have some errors,

no problem with that. It is just interesting what it would be.

> can you copy the raw data

sure
[solar2mqtt_raw_data_POW-HVM10.2M_v7906.txt](https://github.com/user-attachments/files/18049389/solar2mqtt_raw_data_POW-HVM10.2M_v7906.txt)

**POWMR/RAW/Q1**
00 00 00 000 037 030 036 00 00 000 0030 0000 13

**POWMR/RAW/QPIGS**
229.9 49.9 229.9 49.9 1034 0937 010 454 56.20 000 100 0037 00.0 000.0 00.00 00000 00010101 00 00 00000 110

**POWMR/RAW/QPIGS2**
NAK

**POWMR/RAW/QPIRI**
230.0 44.3 230.0 50.0 44.3 0200 0200 48.0 48.0 47.0 56.2 56.2 2 030 100 0 0 1 1 01 0 0 55.0 0 1 48.0 10 44.0

**POWMR/RAW/QET**
NAK

**POWMR/RAW/QPI**
PI30

**POWMR/RAW/QMOD**
L

**POWMR/RAW/QALL**
NAK

**POWMR/RAW/QMN**
VMII-NXPW5KW

**POWMR/RAW/QPIWS**
10000000000000000000000000000000

**POWMR/RAW/QFLAG**
Wrong Length(14), Contact Dev:EakyzDbjuvxgld

**Reply by softwarecrash** [reply]
> > you are in the pre relase variants, that have some errors,
> 
> no problem with that. It is just interesting what it would be.
> 
> > can you copy the raw data
> 
> sure [solar2mqtt_raw_data_POW-HVM10.2M_v7906.txt](https://github.com/user-attachments/files/18049389/solar2mqtt_raw_data_POW-HVM10.2M_v7906.txt)
> 
> **POWMR/RAW/Q1** 00 00 00 000 037 030 036 00 00 000 0030 0000 13
> 
> **POWMR/RAW/QPIGS** 229.9 49.9 229.9 49.9 1034 0937 010 454 56.20 000 100 0037 00.0 000.0 00.00 00000 00010101 00 00 00000 110
> 
> **POWMR/RAW/QPIGS2** NAK
> 
> **POWMR/RAW/QPIRI** 230.0 44.3 230.0 50.0 44.3 0200 0200 48.0 48.0 47.0 56.2 56.2 2 030 100 0 0 1 1 01 0 0 55.0 0 1 48.0 10 44.0
> 
> **POWMR/RAW/QET** NAK
> 
> **POWMR/RAW/QPI** PI30
> 
> **POWMR/RAW/QMOD** L
> 
> **POWMR/RAW/QALL** NAK
> 
> **POWMR/RAW/QMN** VMII-NXPW5KW
> 
> **POWMR/RAW/QPIWS** 10000000000000000000000000000000
> 
> **POWMR/RAW/QFLAG** Wrong Length(14), Contact Dev:EakyzDbjuvxgld

try this
[Solar2MQTT_2.0.0-Pre6A2.zip](https://github.com/user-attachments/files/18050103/Solar2MQTT_2.0.0-Pre6A2.zip)

**Reply by AlexK98** [reply]
2.0.0-Pre6A2 shows 65 sensors, compared to 56 sensors on 2.0.0-Pre6. **Nice**!!!

However there is still unnamed entry with 0 value:
![2024-12-08 20 35 41](https://github.com/user-attachments/assets/4387d462-689c-45b6-84eb-5e8bb02ee668)

New data:
Q1 = 00 00 00 000 036 030 035 00 00 000 0030 0000 13
QPIGS = 230.8 49.9 230.8 49.9 1569 1548 015 454 56.20 000 100 0036 00.0 000.0 00.00 00000 00010101 00 00 00000 110
QPIGS2 = NAK
QPIRI = 230.0 44.3 230.0 50.0 44.3 0200 0200 48.0 48.0 47.0 56.2 56.2 2 030 100 0 0 1 1 01 0 0 55.0 0 1 48.0 10 44.0
QET = NAK
QPI = PI30
QMOD = L
QALL = NAK
QMN = VMII-NXPW5KW
QPIWS = 10000000000000000000000000000000
QFLAG = EakyzDbjuvxgld

**Reply by softwarecrash** [reply]
> 2.0.0-Pre6A2 shows 65 sensors, compared to 56 sensors on 2.0.0-Pre6. **Nice**!!!
> 
> However there is still unnamed entry with 0 value: ![2024-12-08 20 35 41](https://private-user-images.githubusercontent.com/20613948/393633919-4387d462-689c-45b6-84eb-5e8bb02ee668.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM2ODc3NTIsIm5iZiI6MTczMzY4NzQ1MiwicGF0aCI6Ii8yMDYxMzk0OC8zOTM2MzM5MTktNDM4N2Q0NjItNjg5Yy00NWI2LTg0ZWItNWU4YmIwMmVlNjY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA4VDE5NTA1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxMDUwYzU2MmRjY2RhZWNhYjg3MzU2OTY3NzllM2VjMjFlYTIzZTk4MTVlNmE3NGYzZTljZDQ5NGE0YzQ5ZmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Gvnjcd13LQofempV2FqO0_IfrZ865_hyjsKPO8OrYR0)
> 
> New data: Q1 = 00 00 00 000 036 030 035 00 00 000 0030 0000 13 QPIGS = 230.8 49.9 230.8 49.9 1569 1548 015 454 56.20 000 100 0036 00.0 000.0 00.00 00000 00010101 00 00 00000 110 QPIGS2 = NAK QPIRI = 230.0 44.3 230.0 50.0 44.3 0200 0200 48.0 48.0 47.0 56.2 56.2 2 030 100 0 0 1 1 01 0 0 55.0 0 1 48.0 10 44.0 QET = NAK QPI = PI30 QMOD = L QALL = NAK QMN = VMII-NXPW5KW QPIWS = 10000000000000000000000000000000 QFLAG = EakyzDbjuvxgld

ahhh.... there is no name, we dont know what this flag means.
here a updated version with unknown name, when you find out what the flag means we can add the function of it
[Solar2MQTT_2.0.0-Pre6A3_OTA.bin.zip](https://github.com/user-attachments/files/18053797/Solar2MQTT_2.0.0-Pre6A3_OTA.bin.zip)

**Reply by AlexK98** [reply]
> Solar2MQTT_2.0.0-Pre6A3_OTA.bin.zip

Uploaded firmware to my d1 mini.
Yeah, I see that it is now with "Unknown" name.

Interestingly, in MQTT Explorer I see different amount of topics reported. In HA I have 65 topics.
![2024-12-09 01 21 40](https://github.com/user-attachments/assets/c24fae7c-a259-4cc7-b1e8-6238eeeacae1)

And I tried to match topics reported in MQTT explorer with Sensors visible in HA:
![Home Assistant](https://github.com/user-attachments/assets/0b3a90cd-7966-4718-bffd-1eb932b62c6d)

Hope it can help.

**Reply by softwarecrash** [reply]
> > Solar2MQTT_2.0.0-Pre6A3_OTA.bin.zip
> 
> Uploaded firmware to my d1 mini. Yeah, I see that it is now with "Unknown" name.
> 
> Interestingly, in MQTT Explorer I see different amount of topics reported. In HA I have 65 topics. ![2024-12-09 01 21 40](https://private-user-images.githubusercontent.com/20613948/393652444-c24fae7c-a259-4cc7-b1e8-6238eeeacae1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NDAyMDIsIm5iZiI6MTczMzczOTkwMiwicGF0aCI6Ii8yMDYxMzk0OC8zOTM2NTI0NDQtYzI0ZmFlN2MtYTI1OS00Y2M3LWIxZTgtNjIzOGVlZWFjYWUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA5VDEwMjUwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMxMjVjMTViYTM5MjQyOWM5N2M5ZDZiOWMwZjU0ZGY1ZTY0NjIzYzEzMmQ4ZWFlNjYzNmJkMDNjZDdmZmI4OTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.52Ni0vuUgP-fr-4-p626hMgLr4Yx1u0UbRyq9MZmbbU)
> 
> And I tried to match topics reported in MQTT explorer with Sensors visible in HA: ![Home Assistant](https://private-user-images.githubusercontent.com/20613948/393652483-0b3a90cd-7966-4718-bffd-1eb932b62c6d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NDAyMDIsIm5iZiI6MTczMzczOTkwMiwicGF0aCI6Ii8yMDYxMzk0OC8zOTM2NTI0ODMtMGIzYTkwY2QtNzk2Ni00NzE4LWJmZmQtMWViOTMyYjYyYzZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA5VDEwMjUwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMwMzc4OTE1OTVlZGQwYTc0Y2ViMDZhZjk1NTk3OTJkNDUzNGYxODlkODE5ZmQ1MzI0MjliMTI2YWZlMDEyYTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.n7sC-QLAvaMjlbbFSKhfQc8-2O8KptA_5_19espgWiE)
> 
> Hope it can help.

thanks that help, i lost the overview on the datapoints, so actual not all (the new) values are adopted to HomeAssistant discovery, that comes later.

**Reply by softwarecrash** [reply]
@AlexK98 
i hope i get it all
[Solar2MQTT_d1_mini_2.0.0-Pre6A4.zip](https://github.com/user-attachments/files/18219164/Solar2MQTT_d1_mini_2.0.0-Pre6A4.zip)

**Reply by AlexK98** [reply]
@softwarecrash Nice! Thanks.
Generally works like a charm! 72 entities in HA and 74 in MQTT explorer.
I still see "unknown" in MQTT explorer and, as I understand, you did not add to HA "Data_Log_Pop_Up" (can't get any idea what it should do).

As of the moment, I'm charging the battery and set **Current_Max_AC_Charging_Current** to 40A, so the inverter actually show it pushes 39A, same value shows BMS. I suppose **Battery_Charge_Current** in MQTT should show same 39A, but it shows 33-34A, same 33-34A are shown in **Battery_Load** parameter. I will test what these params show during discharging the battery.

```
Q1
00 00 38 000 028 014 022 00 00 000 0100 0000 11
QPIGS
227.0 50.0 227.0 50.0 0703 0703 006 454 53.80 034 095 0028 00.0 000.0 00.00 00000 00010101 00 00 00000 010
QPIGS2
NAK
QPIRI
230.0 44.3 230.0 50.0 44.3 0200 0200 48.0 48.0 42.0 58.4 58.0 2 040 100 0 0 1 1 01 0 0 56.0 0 1 48.0 10 44.0
QET
NAK
QPI
PI30
QMOD
L
QALL
NAK
QMN
VMII-NXPW5KW
QPIWS
10000000000000000000000000000000
QFLAG
EakyzDbjuvxgld
```

**Reply by softwarecrash** [reply]
@AlexK98 yes, the unknown doenst comes to HA, it cant handle states with unknown value, yes i forgot the "Data_Log_Pop_Up" so i have no idea what it is
the current hing is funny.. a lot of devices works like input->charge limit->inverter->battery
so when you set it to 40A and your battery becomes 34A or something, this is correnct, so the inverter self takes 2-8A for working without substracting this from the values for limits.

**Reply by AlexK98** [reply]
> so when you set it to 40A and your battery becomes 34A or something, this is correnct

No, BMS of battery shows 39A, which is actual charging current inverter provides. Same value is shown on inverter display. And this same value is shown in the SmartESS app for the wifi comms dongle.

**Reply by softwarecrash** [reply]
Ah okay, then i have No Idea whats wrong.
So the Data you get IS extracted from the answers, ITS possible that the naming or some Others is wrong

**Reply by AlexK98** [reply]
@softwarecrash, btw, 0200 in QPIRI is most probably 10200, cause my inverter is 10200 rated power.

**Reply by AlexK98** [reply]
Also, I tested a few query commands for PI30 protocol:
- QMCHGCR returns 010 020 030 040 050 060 070 080, but the inverter allows to set values up to 160A for Max Charging Current
- QMUCHGCR returns 002 010 020 030 040 050 060, but the max value my inverter for AC Charging Current is 140A

Seems like it reports values that are appropriate for VMII-NXPW5KW model it reports :(
Maybe there are other ways to get all possible values?

**Reply by softwarecrash** [reply]
the "VMII-NXPW5KW" is was your device spits out on request, there is nothing on the firmware that hold this information or limits.
so all you see is what your device gives you, i can literaly doing nothing about that.
its possible that you have a device that hold modified firmware for a 5kw model, and other hardware arround it to reach 10kw

**Reply by AlexK98** [reply]
Then, the question is how SmartESS knows these charging limit values (160 and 140 respectively) ? Did they simply hardcode them and show to everyone? :)

**Reply by softwarecrash** [reply]
no, they use another protocol that i dont know, so i dont have one of this devvices, i cant rebuild it

**Reply by AlexK98** [reply]
@softwarecrash , do you happen to know what "Warning_Code" value "2" means? I'm seeing it right now.

**Reply by softwarecrash** [reply]
Nope

**Reply by AlexK98** [reply]
@softwarecrash 
Hopefully, A few useful notes on the correlation of values of settings on POW-HVM10.2M. I have no PV modules so I can't test/see other states.

Status_Flag = 10000 -> Inverter_Operation_Mode = Battery, Inverter_Charge_State = No Charging
Status_Flag = 10101 -> Inverter_Operation_Mode = Line, Inverter_Charge_State = either Bulk or Float

Device_Status = 110 -> Inverter_Charge_State = Float
Device_Status = 10 -> Inverter_Charge_State = either No Charging or Bulk

![2024-12-24 15 57 47](https://github.com/user-attachments/assets/6f96ff5a-6165-47d7-b16a-04c660712fc7)

Output_mode or Operation_Logic values did not change during the observation of above settings.

**Reply by AlexK98** [reply]
Also, my inverter responds to QVFW and QVFW2 commands

**Comment by YAYASIMPSON** [comment]
hello sorry now i can't respond you because i have a problem with my inverter i have error 9 and doesn't works since 3 weeks 😔 so i try to repair or change motherboard in waranty and i come here to helo you after.

**Reply by AlexK98** [reply]
Thanks, I already managed to get it working. Hope you find a good repair parts.

**Comment by AlexK98** [comment]
@softwarecrash 
on the latest test firmware (2.0.0-Pre6A4) I found a bug - warning is shown on the dashboard for the disabled entity.

![image](https://github.com/user-attachments/assets/c665ad88-e74e-49e2-932d-26e4ddadc86f)

It does not want to disappear after browser window refresh or even after HA restart

**Reply by AlexK98** [reply]
This 'bug' is only visible on very default HA dashboard.

**Comment by AlexK98** [comment]
@softwarecrash 
I tested values of "Charger source priority" setting and its corresponding PCP commands and actual mode in the inverter:

1. PCP01 => sets 1 => Solar and Utility
2. PCP02 => sets 2 => Only Solar
3. PCP03 => sets 3 => Solar First

PCP00 returns ACK, but does not change active value.

**Comment by AlexK98** [comment]
One more note for POW-HVM10.2M with 79.06 firmware:
![image](https://github.com/user-attachments/assets/24fc2ca4-958f-4126-8f85-ef78f906c16f)

I could not find any references of these values in the Protocols folder.

**Reply by softwarecrash** [reply]
then its a modified answer, ask your seller for the protocol

**Reply by AlexK98** [reply]
@softwarecrash already asked and they refused

**Comment by AlexK98** [comment]
Don't know if this is a bug or not, but despite fans being completely off, the Fan Speed shows 30%
![image](https://github.com/user-attachments/assets/7a876122-d4c5-4b65-a6e9-a65c3e88d34a)

**Reply by softwarecrash** [reply]
its what your device send

**Reply by AlexK98** [reply]
:(

**Reply by dret1985** [reply]
PV2 and PV_TOTAL are displayed to you, if so, can you share the firmware?

**Reply by AlexK98** [reply]
> PV2 and PV_TOTAL are displayed to you, if so, can you share the firmware?

Sorry, these values are not shown in my firmware.
